youtube_dl/jsinterp2/jsinterp.py

```diff
@@ -241,9 +241,16 @@ def interpret_expression(self, expr):
 
         elif name is Token.ID:
             # XXX error handling (unknown id)
-            ref = (self.this[expr[1]] if expr[1] in self.this else
-                   self.global_vars[expr[1]])
-        
+            id = expr[1]
+            try:
+                ref = (self.this[id] if id in self.this else
+                       self.global_vars[id])
+            except KeyError:
```

Comment:
I think `JSInterpreter#extract_function` is useful filtering code before execution, therefore I'd like to continue supporting it. But here it needs to get object from outer context, and that behaviour is **not** in spec. I'd like to suggest a flag that disables these kind of features.